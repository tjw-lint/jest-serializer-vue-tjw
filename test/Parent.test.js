const helpers = require('./helpers.js');

import { mount, shallowMount } from '@vue/test-utils';
import Parent from './components/Parent.vue';

describe('Parent.vue', () => {
  beforeEach(() => {
    helpers.mockSettings({});
  });

  test('Mount snapshot', () => {
    const wrapper = mount(Parent);

    expect(wrapper.html())
      .toMatchSnapshot();

    expect(wrapper.element)
      .toMatchSnapshot();
  });

  test('Shallow snapshot', () => {
    const wrapper = shallowMount(Parent);

    expect(wrapper.html())
      .toMatchSnapshot();
  });

  test('Properly serializes a shallowly-rendered wrapper', () => {
    const wrapper = shallowMount(Parent);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Properly serializes a fully-mounted wrapper', () => {
    const wrapper = mount(Parent);

    expect(wrapper)
      .toMatchSnapshot();
  });
});
