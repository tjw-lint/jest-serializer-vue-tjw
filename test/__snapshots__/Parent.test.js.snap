// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parent.vue Mount snapshot 1`] = `
<div>
  <h1>parent</h1>
  <main>
    <p>this is a child component</p>
    <p>this is a child component</p>
    <p>this is a child component</p>
  </main>
</div>
`;

exports[`Parent.vue Mount snapshot 2`] = `
<div>
  <h1>
    parent
  </h1>
   
  <main>
    <p>
      this is a child component
    </p>
     
    <p>
      this is a child component
    </p>
     
    <p>
      this is a child component
    </p>
  </main>
</div>
`;

exports[`Parent.vue Properly serializes a fully-mounted wrapper 1`] = `
<div>
  <h1>parent</h1>
  <main>
    <p>this is a child component</p>
    <p>this is a child component</p>
    <p>this is a child component</p>
  </main>
</div>
`;

exports[`Parent.vue Properly serializes a shallowly-rendered wrapper 1`] = `
<div>
  <h1>parent</h1>
  <main>
    <child-stub></child-stub>
    <child-stub></child-stub>
    <child-stub></child-stub>
  </main>
</div>
`;

exports[`Parent.vue Shallow snapshot 1`] = `
<div>
  <h1>parent</h1>
  <main>
    <child-stub></child-stub>
    <child-stub></child-stub>
    <child-stub></child-stub>
  </main>
</div>
`;
